doc/nrf-bm/libraries/nrf_sdh.rst

```diff
@@ -3,4 +3,189 @@
+
+The interrupts forwarded to the SoftDevice are declared as Zero Latency interrupts, meaning they cannot be masked using :c:func:`irq_lock`.
+
+Events and observers
+====================
+
+The SoftDevice handler lets the application register functions to receive various events.
+A function that receives SoftDevice handler events is called an observer.
+
+There are four type of observers, each receiving different types of events:
+
+- State observers: events pertaining to the SoftDevice state (enabled/disabled)
+- Bluetooth observers: Bluetooth events coming from the SoftDevice
+- SoC observers: SoC events coming from the SoftDevice
+- Stack observers: All events coming from the SoftDevice
+
+An observer has a priority relative to other observers of the same type.
+The priority of an observer determines the order with which the observers receive the same event.
+
+Five priority levels are defined: ``HIGHEST``, ``HIGH``, ``USER``, ``USER_LOW`` and ``LOWEST``.
+
+An observer may optionally have a context, defined as a pointer, that is passed as a parameter to the observer function.
+
+
+Configuration
+*************
+
+The library is enabled by default in all projects using the SoftDevice.
+
+The SoftDevice handler has several Kconfig options that can be used to configure the clock and the Bluetooth stack.
+
+Dispatch model
+==============
+
+The SoftDevice handler implements SoftDevice event ISR, receiving events from the SoftDevice and dispatching them to the registered observers.
+
+The SoftDevice handler can be configured to dispatch those events in three different ways, using the following Kconfig options:
+
+ # :kconfig:option:`CONFIG_NRF_SDH_DISPATCH_MODEL_IRQ`: Events are dispatched from the SoftDevice event ISR (default)
+ # :kconfig:option:`CONFIG_NRF_SDH_DISPATCH_MODEL_SCHED`: Events are dispatched using the event scheduler library, from the main application loop
+ # :kconfig:option:`CONFIG_NRF_SDH_DISPATCH_MODEL_POLL`: Events are dispatched from the same context as the function calling :c:func:`nrf_sdh_evts_poll`
+
+
+Clock configuration
+===================
+
+The SoftDevice handler has several options to control the clock configuration passed to the :c:func:`sd_softdevice_enable` function.
+
+These options are used automatically when the user enables the SoftDevice by calling the :c:func:`nrf_sdh_enable_request` function.
+
+Bluetooth configuration
+=======================
+
+The SoftDevice Bluetooth stack can store a set of configuration options and associate them with a numerical tag called a connection configuration tag.
+
+The SoftDevice handler has several options that can be used to automatically create a Bluetooth configuration and associate it with the connection configuration tag :kconfig:option:`CONFIG_NRF_SDH_BLE_CONN_CFG_TAG`.
+
+
+Usage
+*****
+
+Declaring observers
+===================
+
```

Comment:
Also here, you need an introductory text. It is not a good practice to have titles one after another without any text between.